### PR TITLE
refactor(tooltip): use hooks

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Refactors the `Tooltip` component to use hooks.

### CODE REVIEW

File: src/components/tooltip/tooltip.js [hunk 1]
```diff
@@ -23,221 +24,230 @@ TooltipWrapper.propTypes = {
...
-    }
-
-    if (this.props.closeAfter) {
-      this.leaveTimer = setTimeout(() => {
-        this.handleClose();
-      }, this.props.closeAfter);
-    } else {
-      this.handleClose();
-    }
+  React.useEffect(() => {
```
Comment (tdeekens) [inline src/components/tooltip/tooltip.js]: Should we build a `useTimeout` for this? Seeems like a frequent case?

File: src/components/tooltip/tooltip.js [hunk 2]
```diff
@@ -23,221 +24,230 @@ TooltipWrapper.propTypes = {
...
+      if (event.type === 'mouseleave' && onMouseLeave) {
+        onMouseLeave(event);
+      }
+
+      if (event.type === 'blur' && onBlur) {
+        onBlur(event);
+      }
+
+      if (closeAfter) {
+        leaveTimer.current = setTimeout(() => {
```
Comment (tdeekens) [inline src/components/tooltip/tooltip.js]: `leaveTimerRef` would have helped me. Wondering if 

```js
const [ triggerTimeout ] = useTimeout(close, 1000) 
```

or something alike could help.

File: src/components/tooltip/tooltip.spec.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
+import { act } from 'react-dom/test-utils';
```
Comment (montezume) [inline src/components/tooltip/tooltip.spec.js]: I've spent so much time battling this over the last week...

Comment (emmenko) [reply src/components/tooltip/tooltip.spec.js]: Hmm interesting...

Comment (montezume) [reply src/components/tooltip/tooltip.spec.js]: this thread is helpful

https://github.com/facebook/react/issues/15379

TL:DR they have added an async act in the react 16.9 which should drop shortly

Comment (emmenko) [reply src/components/tooltip/tooltip.spec.js]: Yeah but RTL already uses `act`. But I guess updating the timers lies "outside" of RTL scope...

Comment (montezume) [reply src/components/tooltip/tooltip.spec.js]: the worst is combining async actions + timers. It's really very hard to test with RTL

**Comment by montezume** [comment]
Can you take a look @jonnybel ?

**Pull request merged**